gum/backend-x86/gumstalker-x86.c

```diff
@@ -2801,14 +2805,20 @@ gum_exec_block_virtualize_branch_insn (GumExecBlock * block,
   else if (op->type == X86_OP_MEM)
   {
 #if GLIB_SIZEOF_VOID_P == 4 && defined (HAVE_WINDOWS)
-    if (ctx->stalker->wow64_transition_address != NULL &&
-        op->mem.disp == GPOINTER_TO_UINT (
-            ctx->stalker->wow64_transition_address) &&
-        op->mem.segment == X86_REG_INVALID &&
+    if (op->mem.segment == X86_REG_INVALID &&
         op->mem.base == X86_REG_INVALID &&
         op->mem.index == X86_REG_INVALID)
     {
-      return gum_exec_block_virtualize_wow64_transition (block, gc);
+      if (ctx->stalker->ntdll_wow64_transition_address != NULL &&
+          op->mem.disp == GPOINTER_TO_UINT (ctx->stalker->ntdll_wow64_transition_address)) 
+      {
+        return gum_exec_block_virtualize_wow64_transition (block, gc, ctx->stalker->ntdll_wow64_transition_address);
+      }
+      else if (ctx->stalker->kernel32_wow64_transition_address != NULL &&
+          op->mem.disp == GPOINTER_TO_UINT (ctx->stalker->kernel32_wow64_transition_address))
+      {
+        return gum_exec_block_virtualize_wow64_transition (block, gc, ctx->stalker->kernel32_wow64_transition_address);
+      }
```

Comment:
Here and above it looks like multiple lines exceed 80 cols.

Reply:
i think i got it right this time, hopefully